src/components/client/TransactionHeader.tsx

```diff
@@ -86,6 +92,20 @@ export const TransactionHeader: React.FC<{
     });
   };
 
+  useEffect(() => {
+    // when rpc endpoints setting change, update the current selected rpc
+    if (rpcEndpoints.map.hasOwnProperty(rpcEndpoint.id)) {
+      if (
```

Comment:
This is kinda brittle coz now we need to remember to update this piece of code if we ever change the data model for `IRPCEndpoint`. 

Do you think this is enough of a corner case where we just go without the second `if`? i.e. if the currently selected endpoint changes, just refresh it regardless?

Reply:
good point, the reason of the second "if" is to avoid unnecessary rpc request on every changes on settings

Reply:
I will rewrite it using shallow() so that it won't matter if the struct changes

Reply:
> good point, the reason of the second "if" is to avoid unnecessary rpc request on every changes on settings

Shouldn't it be only every time the currently selected endpoint is updated?

Reply:
> > good point, the reason of the second "if" is to avoid unnecessary rpc request on every changes on settings
> 
> Shouldn't it be only every time the currently selected endpoint is updated?

The second "if" ensures that it only sends rpc request when the setting of selected rpc endpoint changes

Reply:
> > good point, the reason of the second "if" is to avoid unnecessary rpc request on every changes on settings
> 
> Shouldn't it be only every time the currently selected endpoint is updated?

You are right, we already have done the shallow compare for re-render in the selector. I have remove this extra comparison.